src/westpa/westext/wess/wess_driver.py

```diff
@@ -112,7 +112,15 @@ def prepare_new_iteration(self):
 
         with self.data_manager.lock:
             wess_global_group = self.data_manager.we_h5file.require_group('wess')
+            reweighting_history_dataset = wess_global_group.require_dataset(
+                'reweighting_history', (1,), maxshape=(None,), dtype=int
+            )
             last_reweighting = int(wess_global_group.attrs.get('last_reweighting', 0))
+            if last_reweighting > n_iter:
+                last_reweighting = n_iter - 1
+                reweighting_history = np.array(reweighting_history_dataset)
```

Comment:
I would just change this to `reweighting_history = reweighting_history_dataset[:]` Less of a performance hit.